arch/arm/src/nrf52/nrf52_start.c

```diff
@@ -203,6 +203,10 @@ void __start(void)
   nrf52_enable_profile(true);
 #endif
 
+#if defined(CONFIG_SCHED_IRQMONITOR) || defined(CONFIG_SEGGER_SYSVIEW)
```

Comment:
why not check CONFIG_ARCH_PERF_EVENTS

Reply:
CONFIG_ARCH_PERF_EVENTS is default y so this would be always included. This will be fixed in the next step with https://github.com/apache/nuttx/pull/10828

Reply:
> CONFIG_ARCH_PERF_EVENTS is default y so this would be always included. This will be fixed in the next step with #10828

let's change the default value to n if you like

Reply:
I think we still need an additional flag indicating that we have to include `up_perf` support. With https://github.com/apache/nuttx/pull/10828 we can select if we want arch specific `up_perf` with `CONFIG_ARCH_PERF_EVENTS=y` or `drivers/timers/arch_timer.c` implementation when `CONFIG_ARCH_PERF_EVENTS=n`.
If we set `CONFIG_ARCH_PERF_EVENTS` default to n and we make including the code dependent on it, then we lose the ability to choose the  `drivers/timers/arch_timer.c` implementation.

Reply:
but CONFIG_ARCH_PERF_EVENTS=n, arch_timer.c will provide implementation of up_perf.

Reply:
Can we merge it as it is now ? This logic is also present in stm32, so it'll be better to change everything at once. I prefer to solve this in a separate PR after https://github.com/apache/nuttx/pull/10834/ merged

Reply:
Sure.